### PR TITLE
WIP : Documentation on Arrow and ZArrow - DON'T MERGE

### PR DESCRIPTION
**Pull request by tampler**

### CODE REVIEW

**Comment by tampler** [comment]
@adamgfraser Pls check this out

File: docs/about/Arrow.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* `Function1[A,B]`
+```scala mdoc:silent
+val plusOne: Int => Int = _ + 1
+val mulTwo: Int => Int  = _ * 2
+
+val composed: Int => Int = plusOne.andThen(mulTwo)
+println(composed(2)) // 6
+```
+
+* Kleisli:  `A => M[B]`, where M is a Monad
```
Comment (adamgfraser) [inline docs/about/Arrow.md]: Maybe to make this more accessible we should start by just calling these effectual functions and then below we can say that in the functional programming literature where the effect is a monad it is called a Kleisli arrow.

File: docs/intro.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Introduction
+`ZIO-Arrow` or `ZArrow` is a high-performance composition effect for ZIO Ecosystem. <br>
+`ZArrow` is an implementation of Hughes[1,2] and Paterson[3] `Arrow` structure for ZIO Effects. 
+
+`ZArrow` delivers three main capabilities:
+
+* High Performance - `ZArrow` exploits `JVM` internals to dramatically decrease the number of allocations and dispatches, yielding an unprecedental runtime performance.
+
+* Abstract interface - `Arrow` is a more abstract data type, than ZIO Monad. It's more abstract than ZIO Streams. In a nutshell, `ZArrow` allows function-like interface which can have both different input and different outputs.
```
Comment (adamgfraser) [inline docs/intro.md]: I think we have more work to do to prove this out. 😃

File: docs/intro.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  (add1 *** mul2).run((3, 3))     // add1 split mul2 == (4, 6)
+  (add1 &&& mul2).run(3)          // add1 merge mul2 == (4 -> 6))
+  (add1 <*> mul2)(_ -> _).run(3)) // add1 zipWith mul2 == (4 -> 6))
+
+  // Complex composition 
+  val composed:Int => Int = (add1 >>> mul2) >>> (add1 <*> mul2)
+  val res = composed.run(1) // (5,8)
+  ```
+
+Looking at those examples, we can conclude the following properties of Arrow Programming:
```
Comment (adamgfraser) [inline docs/intro.md]: Great overview! Do you want to talk at all about when arrows may not be appropriate? I think the general impression is that programming with arrows is less ergonomic, at least for programs that can't be modeled as straightforward data flows. For example if you look at the implementation of bubble sort in the benchmarks it looks much less clear when written in terms of arrows than traditional effects.

Comment (tampler) [reply docs/intro.md]: Arrows in that bubble-sort example are not ergonomic cos they are simply misused there. The big idea of Arrows is to provide a `unified compositional interface` for heterogeneous  data types from different libraries.

I'll cover that in a doc. Simply put, `ZArrow` works well only when it is used appropriately

File: docs/about/Arrow.md (hunk 1 above)
Comment (tampler) [reply docs/about/Arrow.md]: Fixed

File: docs/intro.md (hunk 2 above)
Comment (tampler) [reply docs/intro.md]: Added proofs as benchmarks and JDG LabdaConf 2018 presentation

File: docs/intro.md (hunk 3 above)
Comment (tampler) [reply docs/intro.md]: I'll elaborate more on that later. Let's move gradually
Some documentation is better than its total absence

**Pull request closed without merging**